src/main/java/com/github/fge/jsonpatch/AddOperation.java

```diff
@@ -65,78 +65,62 @@
  *     [ 1, 2, 3 ]
  * </pre>
  */
-public final class AddOperation
-    extends PathValueOperation
-{
-    private static final ReferenceToken LAST_ARRAY_ELEMENT
-        = ReferenceToken.fromRaw("-");
+public final class AddOperation extends PathValueOperation {
+    public static final String LAST_ARRAY_ELEMENT_SYMBOL = "-";
 
     @JsonCreator
-    public AddOperation(@JsonProperty("path") final JsonPointer path,
-        @JsonProperty("value") final JsonNode value)
-    {
+    public AddOperation(@JsonProperty("path") final String path,
+                        @JsonProperty("value") final JsonNode value) {
         super("add", path, value);
     }
 
     @Override
-    public JsonNode apply(final JsonNode node)
-        throws JsonPatchException
-    {
+    public JsonNode apply(final JsonNode node) throws JsonPatchException {
         if (path.isEmpty())
             return value;
-
         /*
          * Check the parent node: it must exist and be a container (ie an array
          * or an object) for the add operation to work.
          */
-        final JsonNode parentNode = path.parent().path(node);
-        if (parentNode.isMissingNode())
-            throw new JsonPatchException(BUNDLE.getMessage(
-                "jsonPatch.noSuchParent"));
+        final int lastSlashIndex = path.lastIndexOf('/');
```

Comment:
Can we have a test with paths like
/this/is/my/path
and
/this/is/my/path/

Will both work?

Reply:
First path obviously works.
Technically the second path leads to an element with name equal to empty string, inside a node with name "path", this is currently not working in json path anyway, I filed a ticket for it and it's on our channel.
Our implementation might work with it, but tests are commented out due to that bug.